### PR TITLE
Fix value for metricsBindAddress field for kapp-controller to be consistent with the type declared in openAPI schema

### LINKED ISSUE #1503: Fixing type for metricsBindAddress field for kapp-controller after defining openAPI schema
**Bug description**
We have schema defined for kapp-controller [Link](https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml#L34), where field metricsBindAddress is defined as string. In framework, we are using int to override it [kapp-controller_addon_data.lib.yaml](https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/ytt/02_addons/kapp-controller/kapp-controller_addon_data.lib.yaml#L28) is failing unit tests now because it is checking it against schema.
Earlier the tests were not failing because it ends up to be "stringified" [here](https://github.com/vmware-tanzu/community-edition/blob/0054c7f86937a0cfb9bd6dbc7501128fc06d4194/addons/packages/kapp-controller/0.28.0/bundle/config/overlays/update-deployment.yaml#L23) 

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [X] Test and Release
- [ ] User Experience

**Expected behavior**
Unit tests for example : https://github.com/vmware-tanzu/tanzu-framework/runs/4811561957?check_suite_focus=true, should not fail

### PR DESCRIPTION
**Pull request by shivaani0505**
### What this PR does / why we need it
Fix value for metricsBindAddress field for kapp-controller to be consistent with the type declared in openAPI schema

### Which issue(s) this PR fixes

Fixes #1503 

### Describe testing done for PR

Tested by generating manifests

### Release note

```release-note
Fix value for metricsBindAddress field for kapp-controller to be consistent with the type declared in openAPI schema
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by blc1996 — APPROVED** [review]
LGTM

**Comment by alfredthenarwhal** [comment]
> @shivaani0505:  /test
Commit: edc08cc

Tests failed! Build no: 1369

**Comment by alfredthenarwhal** [comment]
> @navidshaikh:  /test install-vc7
Commit: edc08cc

Tests failed! Build no: 1382

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ kappController:
   deployment:
     #! The coreDNSIP will be injected into /etc/resolv.conf of kapp-controller pod
     coreDNSIP: null
-    hostNetwork: null
+    hostNetwork: false
```
Comment (danniel1205) [inline pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml:12]: question: I think we are using `hostNetwork:true` with `dnsPolicy:ClusterFirst` for kapp-controller. If assigning `false` is intended, could you please help put some comments ? thanks

Comment (blc1996) [reply pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml:12]: This is the default value of the package template. The value we use in addon secret is located else where.

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    coreDNSIP: ""
+    #@schema/desc "Whether to enable host networking for kapp-controller deployment"
+    hostNetwork: false
+    #@schema/desc "The priority value that various system components use to find the priority of the kapp-controller pod"
+    #@schema/nullable
+    priorityClassName: ""
+    #@schema/desc "Concurrency of kapp-controller deployment"
+    concurrency: 4
+    #@schema/desc "kapp-controller deployment tolerations"
+    tolerations: ["toleration1"]
```
Comment (danniel1205) [inline pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:29]: suggestion: toleration seems not an array of string, not sure how to define an array of struct/map in schema.

```
      tolerations:
      - key: CriticalAddonsOnly
        operator: Exists
      - effect: NoSchedule
        key: node-role.kubernetes.io/master
      - effect: NoSchedule
        key: node.kubernetes.io/not-ready
      - effect: NoSchedule
        key: node.cloudprovider.kubernetes.io/uninitialized
        value: "true"
```

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml (hunk 1 above)
Comment (shivaani0505) [reply pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/values.yaml:12]: The value with which we override is located in https://github.com/vmware-tanzu/tanzu-framework/blob/main/pkg/v1/providers/ytt/02_addons/kapp-controller/kapp-controller_addon_data.lib.yaml#L25
We are not changing that.It is `true`

File: pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml (hunk 2 above)
Comment (shivaani0505) [reply pkg/v1/providers/ytt/vendir/kapp-controller/_ytt_lib/addons/packages/kapp-controller/0.30.0/bundle/config/schema.yaml:29]: I think we should be able to use annotation:
`#@schema/type any=True` as per
https://carvel.dev/ytt/docs/latest/how-to-write-schema/#allowing-multiple-types-of-maps-or-arrays

I will add above for `tolerations` key

**Comment by alfredthenarwhal** [comment]
> @shivaani0505:  /test install-vc7
Commit: edc08cc

Tests failed! Build no: 1384

**Comment by blc1996** [comment]
After some discussion with @shivaani0505 this morning, I recommend closing this MR. We don't need to port the schema changes into tanzu-framework as for now. It can be done later when all the moving parts have been finalized.

**Comment by shivaani0505** [comment]
Closing as per [comment](https://github.com/vmware-tanzu/tanzu-framework/pull/1526#issuecomment-1018014483)

**Pull request closed without merging**